### DISCUSSION TITLE
ojs inputs to R

### DISCUSSION THREAD

**Discussion by actuariat** [quarto-dev/quarto-cli#612] [Q&A] [answered]
I want to pass data collected by ojs (dataojs), like [ viewof dataojs = Inputs.range(..) ] to R like define_ojs does from R to ojs. 

This is crucial: but I didn't find a single mention or something.

**Comment by cscheid** [comment]
Sorry that the information is somewhat hard to find! The explanation is here: https://quarto.org/docs/interactive/ojs/shiny.html By default, `viewof`s are automatically propagated to shiny, so you should be able to just refer to them by name.

Edit: here's a minimal example on shinyapps: https://jjallaire.shinyapps.io/kmeans-shiny-ojs/

**Reply by actuariat** [reply]
Thank you for the quick answer. Sorry I wasn't clear myself. I want to replace shiny with ojs. So no shiny server, but simply to use ojs instead to gather data in an interactive way, pass it to R, make calculations and then show the results. 

Your example it is quite confusing, all I get it is that I need to use **input.x** in order to get the collected **x** from ojs.  But again, I do not want to use reactive (no connection to shiny at all). 

Of course one can use shiny (no need for ojs), but the idea is to replace shiny in order to **_have a fully self-contained presentation_** that works locally without any server running.

So please provide a very simple .qmd with one simple rule: passing one interactive input to R. I couldn't do it, because I do not get it how ojs works.

**Reply by cscheid** [reply, marked as answer]
Unfortunately, I think that is not possible with the current version of R + ojs. If you have an R function whose values depend on an OJS interactive input, then you will need that R function to execute every time the interactive input changes. As a result, you need shiny to host the execution of that R function.
